core/server/update-check.js

```diff
@@ -161,16 +163,16 @@ function updateCheckResponse(response) {
         api.settings.edit(
             {settings: [{key: 'nextUpdateCheck', value: response.next_check}]},
             internal
-        ).catch(errors.rejectError),
+        ).catch(errors.rejectError).reflect(),
         api.settings.edit(
             {settings: [{key: 'displayUpdateNotification', value: response.version}]},
             internal
-        ).catch(errors.rejectError)
+        ).catch(errors.rejectError).reflect()
     );
 
-    return Promise.settle(ops).then(function then(descriptors) {
+    return Promise.map(ops).then(function then(descriptors) {
```

Comment:
Hi :wink: 

`Promise.props` has the same behavior.  Given ` obj = { p1: rejectedPromise1, p2: rejectedPromise2 }`, the result of `Promise.props(obj)` will be a rejected promise for `rejectedPromise1`.